src/System Application/App/AppSource Gallery/src/AppSourceProductDetails.Page.al

```diff
@@ -0,0 +1,395 @@
+                }
+                field(Links_SupportUri; AppSourceProductManager.GetStringValue(ProductObject, 'supportUri'))
+                {
+                    Caption = 'Support Uri';
+                    ToolTip = 'Specifies the Support Uri';
+                    ExtendedDatatype = Url;
+                }
+            }
+        }
+    }
+
+    actions
+    {
+        area(Promoted)
+        {
+            actionref(Open_Promoted; OpenInAppSource) { }
+            actionref(Install_Promoted; Install) { }
+            actionref(Uninstall_Promoted; Uninstall) { }
+        }
+
+        area(Processing)
+        {
+            action(OpenInAppSource)
+            {
+                Caption = 'View in AppSource';
+                Scope = Page;
+                Image = Open;
+                ToolTip = 'View app in AppSource';
+
+                trigger OnAction()
+                begin
+                    AppSourceProductManager.OpenInAppSource(UniqueProductID);
+                end;
+            }
+
+            action(Install)
+            {
+                Caption = 'Install App';
+                Scope = Page;
+                Enabled = CurrentRecordCanBeInstalled;
+                Image = Insert;
+                ToolTip = 'Install App';
+
+                trigger OnAction()
+                var
+                    ExtensionManagement: Codeunit "Extension Management";
+                begin
+                    if not (CurrentRecordCanBeInstalled) then
+                        exit;
+
+                    if (PlansAreVisible) then
+                        if not Confirm(PurchaseLicensesElsewhereLbl) then
+                            exit;
+                    ExtensionManagement.InstallMarketplaceExtension(AppID);
+                end;
+            }
+
+            action(Uninstall)
+            {
+                Caption = 'Uninstall App';
+                Scope = Page;
+                Enabled = CurrentRecordCanBeUninstalled;
+                Image = Delete;
+                ToolTip = 'Uninstall App';
+                AccessByPermission = TableData "Installed Application" = d;
+
+                trigger OnAction()
+                begin
+                    ExtensionManagement.UninstallExtension(AppID, true);
+                end;
+            }
+        }
+    }
+
+    var
+        ExtensionManagement: Codeunit "Extension Management";
+        AppSourceProductManager: Codeunit "AppSource Product Manager";
+        ProductObject: JsonObject;
+        UniqueProductID: Text;
+        AppID: Text;
+        CurrentRecordCanBeUninstalled: Boolean;
+        CurrentRecordCanBeInstalled: Boolean;
+        PlansOverview: Text;
+        PlansAreVisible: Boolean;
+        PurchaseLicensesElsewhereLbl: Label 'Installing this app might lead to undesired behavior if licenses are not purchaed before use. You must purchase licenses through Microsoft AppSource.\Do you want to continue with the installation?';
```

Comment:
Is there a specific reason why the labels are all public? If a label is just used once, I find it way better if it is declared locally, just as any other variable would be.

Reply:
To address in this PR

Reply:
This is a legacy standard. All text constants (and now labels) have and are always added as globals.

For now, _(and in this PR)_ I think we should stick with the legacy standard but I do think it's a topic we could discuss and see if we should change how we declare labels.

Reply:
FYI @nikolakukrika

Reply:
Good with me :)